src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -577,22 +585,22 @@ private bool CheckIfPathIsNested(ReadOnlySpan<char> eventPath)
                 return _includeChildren || _fullDirectory.AsSpan().StartsWith(System.IO.Path.GetDirectoryName(eventPath), StringComparison.OrdinalIgnoreCase);
             }
 
-            private unsafe int FindRenameChangePairedChange(
+            private unsafe int? FindRenameChangePairedChange(
                 int currentIndex,
-                FSEventStreamEventFlags* eventFlags,
-                int numEvents)
+                Span<FSEventStreamEventFlags> flags, Span<FSEventStreamEventId> ids)
             {
-                // Start at one past the current index and try to find the next Rename item, which should be the old path.
-                for (int i = currentIndex + 1; i < numEvents; i++)
-                {
-                    if (IsFlagSet(eventFlags[i], FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
-                    {
-                        // We found match, stop looking
-                        return i;
-                    }
-                }
+                var nextIndex = currentIndex + 1;
+
+                if (nextIndex >= flags.Length)
+                    return null;
+
+                var currentEvent = (flags: flags[currentIndex], id: ids[currentIndex]);
+                var nextEvent = (flags: flags[nextIndex], id: ids[nextIndex]);
```

Comment:
Why are these tuples helpful here? It'd be clearer IMO (and less expensive) to just do:
```C#
if (ids[currentIndex] + 1 == ids[nextIndex] &&
    IsFlagSet(flags[nextIndex], FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
{
    return nextIndex;
}
```
currentEvent.flags isn't used anywhere, either.

Reply:
The idea was to reduce index mistake by correctly name the specific event.  I was about to pass the currentEvent as an argument for the FindRenameChangePairedChange, but I consider this as unnecessary refactoring of the ProcessEvents method (it was refactored a lot anyway).  I accepted your proposal.